WordPress/gutenberg
# [RNMobile] File and Audio block simplify media flow

Gutenberg Mobile PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/3399


## Description

Adds the media options auto opening behavior to the File and Audio block [that was implemented for the Image, Video, and Gallery Block](https://github.com/WordPress/gutenberg/pull/29546)
## How has this been tested?

### Manual Testing

1. Tap (+) (inserter) – this shows the Block Library sheet
2. Tap Audio and File to add the block's media placeholder to canvas – which triggers the media upload options sheet. 
3. Notice the media upload options auto open.

### E2E Testing

The auto-opening wasn't tested but the tests were modified to dismiss the picker when it opens. The changes follow the same pattern used in the previous PR. To test : 

`TEST_RN_PLATFORM=android/ios npm run native device-tests:local gutenberg-editor-file-@canary.test`

`TEST_RN_PLATFORM=android/ios npm run native device-tests:local gutenberg-editor-audio.test`

### Regression Testing

I noticed that the logic for the video and image block contained redundant logic within the `autoOpenMediaUpload` prop. For both blocks, the media placeholder is not shown if any of the values ( src or url )  are undefined. 66acdbe

1. Tap (+) (inserter) – this shows the Block Library sheet
2. Tap Video and Image to add the block's media placeholder to canvas – which triggers the media upload options sheet. 
3. Notice the media upload options auto open.

## Screenshots 

### File Block 

IOS | Android 
--------|-------
 <kbd><img src="https://user-images.githubusercontent.com/1509205/115495462-492f5980-a22d-11eb-8a0e-2dc897e71366.gif" width="320"></kbd>   |     <kbd><img src="https://user-images.githubusercontent.com/1509205/115495386-21d88c80-a22d-11eb-921d-53f99d48efe3.gif" width="320"></kbd>
  
  ### Audio Block 
  
  IOS | Android 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/115495654-be9b2a00-a22d-11eb-9dcf-d9970e2f4bd3.gif" width="320"></kbd>   |       <kbd><img src="https://user-images.githubusercontent.com/1509205/115495661-c4910b00-a22d-11eb-9b43-c90d18925f71.gif" width="320"></kbd>

## Checklist:
